bwapi/Library/BWAPILIBUnitTest/UnitTypeTest.cpp

```diff
@@ -0,0 +1,839 @@
+}
+
+TEST(Protoss_Observatory, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Protoss_Observatory.whatBuilds(), std::make_pair(UnitTypes::Protoss_Probe, 1));
+}
+
+TEST(Protoss_Gateway, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Protoss_Gateway.whatBuilds(), std::make_pair(UnitTypes::Protoss_Probe, 1));
+}
+
+TEST(Protoss_Photon_Cannon, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Protoss_Photon_Cannon.whatBuilds(), std::make_pair(UnitTypes::Protoss_Probe, 1));
+}
+
+TEST(Protoss_Citadel_of_Adun, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Protoss_Citadel_of_Adun.whatBuilds(), std::make_pair(UnitTypes::Protoss_Probe, 1));
+}
+
+TEST(Protoss_Cybernetics_Core, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Protoss_Cybernetics_Core.whatBuilds(), std::make_pair(UnitTypes::Protoss_Probe, 1));
+}
+
+TEST(Protoss_Templar_Archives, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Protoss_Templar_Archives.whatBuilds(), std::make_pair(UnitTypes::Protoss_Probe, 1));
+}
+
+TEST(Protoss_Forge, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Protoss_Forge.whatBuilds(), std::make_pair(UnitTypes::Protoss_Probe, 1));
+}
+
+TEST(Protoss_Stargate, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Protoss_Stargate.whatBuilds(), std::make_pair(UnitTypes::Protoss_Probe, 1));
+}
+
+TEST(Special_Stasis_Cell_Prison, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Special_Stasis_Cell_Prison.whatBuilds(), std::make_pair(UnitTypes::None, 0));
+}
+
+TEST(Protoss_Fleet_Beacon, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Protoss_Fleet_Beacon.whatBuilds(), std::make_pair(UnitTypes::Protoss_Probe, 1));
+}
+
+TEST(Protoss_Arbiter_Tribunal, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Protoss_Arbiter_Tribunal.whatBuilds(), std::make_pair(UnitTypes::Protoss_Probe, 1));
+}
+
+TEST(Protoss_Robotics_Support_Bay, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Protoss_Robotics_Support_Bay.whatBuilds(), std::make_pair(UnitTypes::Protoss_Probe, 1));
+}
+
+TEST(Protoss_Shield_Battery, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Protoss_Shield_Battery.whatBuilds(), std::make_pair(UnitTypes::Protoss_Probe, 1));
+}
+
+TEST(Special_Khaydarin_Crystal_Form, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Special_Khaydarin_Crystal_Form.whatBuilds(), std::make_pair(UnitTypes::None, 0));
+}
+
+TEST(Special_Protoss_Temple, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Special_Protoss_Temple.whatBuilds(), std::make_pair(UnitTypes::None, 0));
+}
+
+TEST(Special_XelNaga_Temple, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Special_XelNaga_Temple.whatBuilds(), std::make_pair(UnitTypes::None, 0));
+}
+
+TEST(Resource_Mineral_Field, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Resource_Mineral_Field.whatBuilds(), std::make_pair(UnitTypes::None, 0));
+}
+
+TEST(Resource_Mineral_Field_Type_2, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Resource_Mineral_Field_Type_2.whatBuilds(), std::make_pair(UnitTypes::None, 0));
+}
+
+TEST(Resource_Mineral_Field_Type_3, whatBuilds) {
+    EXPECT_EQ(UnitTypes::Resource_Mineral_Field_Type_3.whatBuilds(), std::make_pair(UnitTypes::None, 0));
+}
```

Comment:
I think we should use [parameterized tests](https://google.github.io/googletest/advanced.html#value-parameterized-tests) for these but maybe I'll accept this as-is and refactor it since we don't have any examples of that yet, then you can use the example for future tests.